rust/xaynet-analytics/src/database/common.rs

```diff
@@ -1,35 +1,57 @@
 use anyhow::{anyhow, Error, Result};
 use isar_core::{
     index::StringIndexType,
-    object::{data_type::DataType, object_builder::ObjectBuilder},
+    object::{
+        data_type::DataType,
+        isar_object::{IsarObject, Property},
+        object_builder::ObjectBuilder,
+    },
     schema::collection_schema::CollectionSchema,
 };
-use std::vec::IntoIter;
+use std::{convert::TryFrom, vec::IntoIter};
 
 use crate::database::isar::IsarDb;
 
-pub trait IsarAdapter: Sized {
+pub trait IsarAdapter<'object>: Sized {
     fn into_field_properties() -> IntoIter<FieldProperty>;
 
     fn write_with_object_builder(&self, object_builder: &mut ObjectBuilder);
 
-    fn read(bytes: &[u8]) -> Self;
+    fn read(
+        isar_object: &'object IsarObject,
+        isar_properties: &'object [(String, Property)],
+    ) -> Result<Self, Error>;
+
+    fn find_property_by_name(
+        name: &str,
+        isar_properties: &[(String, Property)],
+    ) -> Result<Property, Error> {
+        let property = isar_properties
+            .iter()
+            .find(|(isar_property_name, _)| isar_property_name == name);
```

Comment:
Here we can remove the let bind and continue after the `find` with `map(...).ok_or_else(|| anyhow!(...))`

Reply:
noice. done